detect-nuget-inspector/detect-nuget-inspector/Inspection/Inspectors/SolutionInspector.cs

```diff
@@ -106,7 +121,31 @@ public Container GetContainer()
                         try
                         {
                             string projectRelativePath = project.Path;
-                            string projectPath = PathUtil.Combine(solutionDirectory, projectRelativePath);
+                            string projectPath = null;
+                            Boolean directoryPackagesExists = false;
+                            if (projectRelativePath.Contains("Directory.Packages.props"))
+                            {
+                                string parentPath = solutionDirectory;
+                                bool fileNotFound = true;
+                                while (fileNotFound)
+                                {
+                                    parentPath =
+                                        parentPath.Substring(0, parentPath.LastIndexOf("/"));
```

Comment:
Should we be taking backslashes into account in Windows cases?

Reply:
Yes, I believe we should be I am updating it to take account for Windows, I do not know if checking Operating System is a viable option though, do you have any suggestions?

Reply:
Looks like we could utilize the Path class a bit more: https://learn.microsoft.com/en-us/dotnet/api/system.io.path?view=net-7.0

For example, the `GetDirectoryName` method returns the parent directory and `GetPathRoot` will give you values such as `C:\` and `/` depending on the platform.

Reply:
I found a way to determine the OperatingSystem https://learn.microsoft.com/en-us/dotnet/api/system.operatingsystem?view=net-8.0 for getting OS information